src/content/tools/hooks.md

```diff
@@ -0,0 +1,294 @@
+---
+title: Hooks
+description: Run custom build scripts.
+---
+
+:::note
+Support for build hooks is in **preview** and
+can be used on the `main` and `beta` channels.
+:::
+
+This guide explains what hooks are and how to use them
+with a package.
+
+## Introduction
+
+You can currently use hooks to do things
+such as compile or download native assets (code written in
+other languages that are compiled into machine code), and
+then call these assets from the Dart code of a package.
+
+Hooks are Dart scripts placed in the `hook/` directory of
+your Dart package. They have a predefined format for their
+input and output, which allows the Dart SDK to:
+
+1. Discover the hooks.
+1. Execute the hooks with the necessary input.
+1. Consume the output produced by the hooks.
+
+Example project with a build hook:
+
+```plaintext
+📁  example_project                   // project with hooks
+    📁  hook                          // add hook scripts here
+        📄 build.dart
+    📁  lib                           // reference your assets here
+        📄 example.dart
+    📁  src                           // add native sources here
+        📄 example_native_library.c
+        📄 example_native_library.h
+    📁  test                          // test your assets here
+        📄 example_test.dart
+```
+
+## Hooks
+
+Currently, the build hook is available, but more are
+planned. To learn more, see the following.
+
+### Build hooks {:.no_toc}
+
+With build hooks, a package can do things such as
+compile or download native assets such as C or Rust libraries.
+Afterwards, these assets can be called from the Dart code of
+a package.
+
+A package's build hook is automatically invoked by the
+Dart SDK at an appropriate time during the build process.
+Build hooks are run in parallel with Dart compilation and
+might do longer running operations such as downloading or
+calling a native compiler.
+
+Use the [`build`][] function to parse the hook
+input with [`BuildInput`] and then write the hook output
+with [`BuildOutputBuilder`]. The hook should place
+downloaded and generated assets in
+[`BuildInput.sharedOutputDirectory`][].
+
+The assets produced for your package might depend on
+[`assets`][] or [`metadata`][] produced by the build hooks
+from the packages in your direct dependencies. Therefore,
+build hooks are run in the order of dependencies.
```

Comment:
The order of what dependencies? From a package's pubspec file, declared in the hook output?

Reply:
Good catch! The pubspec dependencies.